src/applications/caregivers/components/FormFieldsets/AddressWithAutofill.jsx

```diff
@@ -0,0 +1,229 @@
+    event => {
+      formData.autofill = event.target.checked;
+      // transform and send data back to the form
+      const dataToSend = formData.autofill
+        ? { ...formData, ...veteranAddress }
+        : { autofill: false };
+      addDirtyField('autofill');
+      onChange(dataToSend);
+    },
+    [addDirtyField, formData, onChange, veteranAddress],
+  );
+
+  // define our non-checkbox input change event
+  const handleChange = useCallback(
+    event => {
+      const fieldName = event.target.name.split('_').pop();
+      formData[fieldName] = event.target.value;
+      // uncheck autofill since we have modified the input value
+      if (formData.autofill) formData.autofill = false;
+      // send updated date to the form
+      addDirtyField(fieldName);
+      onChange(formData);
+    },
+    [addDirtyField, formData, onChange],
+  );
+
+  // define our non-checkbox input blur event
+  const handleBlur = useCallback(
+    event => {
+      const fieldName = event.target.name.split('_').pop();
+      addDirtyField(fieldName);
+    },
+    [addDirtyField],
+  );
+
+  // check field for validation errors
+  const showError = field => {
+    const errorList = errorSchema[field].__errors;
+    const fieldIsDirty = dirtyFields.includes(field);
+    // validate only if field is dirty or form has been submitted
+    if ((submitted || fieldIsDirty) && errorList.length) {
+      // validate required fields
+      if (reqFields.includes(field) && !formData[field]) {
+        return errorMessages[field].required;
+      }
+      // validate fields with required pattern matches
+      if (schemaProps[field].pattern) {
+        const regex = new RegExp(schemaProps[field].pattern);
+        if (!regex.test(formData[field])) {
+          return errorMessages[field].pattern;
+        }
+      }
+    }
+    return false;
+  };
+
+  return reviewMode ? (
+    <AddressWithAutofillReviewField
+      canAutofillAddress={canAutofillAddress}
+      formData={formData}
+      inputLabel={inputLabelMap[props.name]}
+    />
+  ) : (
+    <fieldset className="cg-address-with-autofill vads-u-margin-y--2">
+      <legend className="vads-u-visibility--screen-reader">
+        {inputLabelMap[props.name]} current address
+      </legend>
+
+      {canAutofillAddress && (
+        <VaCheckbox
+          id={idSchema.autofill.$id}
+          checked={formData.autofill}
+          label="Use the same address as the Veteran"
+          className="vads-u-margin-left--neg3"
+          style={{ marginLeft: '-24px' }}
+          onVaChange={handleCheck}
+        />
+      )}
+
+      <VaTextInput
+        id={idSchema.street.$id}
+        name={idSchema.street.$id}
+        value={formData.street}
+        label={`${inputLabelMap[props.name]} current street address`}
+        className="cg-address-input"
+        error={showError('street') || null}
+        required
+        onInput={handleChange}
+        onBlur={handleBlur}
+      />
```

Comment:
This [web-component accepts an `autocomplete` property](https://design.va.gov/storybook/?path=/docs/components-va-text-input--default), which might be nice to include for auto-prefilling from the Veteran's browser.

Reply:
Will definitely take this back to the team for a future update. We haven't talked about using that property.